Android Initial Tozstore Features

willmichael opened the pull request:
This current state of trunk is relatively untouched boiler plate code from `flutter create --template=package`.

willmichael committed:
add initial TozStore features

willmichael committed:
Remove workspace file

willmichael committed:
cleanup old deps

willmichael committed:
rename and centralize model file

willmichael committed:
Fix and test share/revoke methods

galxy25 commented on android/.project:
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>plugin_tozny</name>
+	<comment>Project plugin_tozny created by Buildship.</comment>
```
~ `created by the friendly folks at Tozny`

efabens commented on android/build.gradle:
```diff
@@ -30,4 +33,29 @@ android {
     }
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_7
+        targetCompatibility JavaVersion.VERSION_1_7
```
lets make these VERSION_1_8 I believe that is best practices and allows some nice language features

willmichael replied:
Agreed this was debugging code that I let slip through

efabens commented on android/build.gradle:
```diff
@@ -30,4 +33,29 @@ android {
+            maven { url "https://dl.bintray.com/terl/lazysodium-maven" }
+        }
+
+        implementation ('com.tozny.e3db:e3db-client-android:5.0.1@aar') {
```
Is there a reason this isn't using 6.0.0?

willmichael replied:
Nope will move to 6.0.0 for identity features

galxy25 commented on android/src/main/java/com/example/plugin_tozny/E3dbSerializer.java:
```diff
@@ -0,0 +1,50 @@
+package com.example.plugin_tozny;
```
prefer `TozStore` or Tozny to E3db

willmichael replied:
ah the "example" slipped in, will refactor that out -- but I do prefer the namespace plugin_tozny even though its ugly... This is not a true Dart SDK which we may want in the future and it doesn't fit in TozStore since it will also have TozID, thoughts?

galxy25 replied:
Sounds like a good case for the generic `Tozny`

galxy25 commented on android/src/main/java/com/example/plugin_tozny/PluginToznyPlugin.java:
```diff
@@ -17,23 +22,188 @@
+      }
+  }
+
+  public void writeRecord(@NonNull MethodCall call, @NonNull final io.flutter.plugin.common.MethodChannel.Result result) {
```
if its not too much, can we get writeFileRecord in this PR?

galxy25 commented on example/android/.project:
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>android</name>
+	<comment>Project android created by Buildship.</comment>
```
`by Tozny`

willmichael replied:
I believe these are auto generated gradle files... but from googling they seem to be part of Eclipse settings which i don't have ❓

galxy25 commented on example/android/app/.project:
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>app</name>
+	<comment>Project app created by Buildship.</comment>
```
or maybe buildship is a build tool? If so :homer-back-away:

galxy25 commented on example/lib/main.dart:
```diff
@@ -42,15 +45,71 @@ class _MyAppState extends State<MyApp> {
+              ),
+            ),
+            Text('Running on: $_platformVersion\n'),
+          ]
```
Tozny reference / test app off to a good start

galxy25 commented on lib/tozny_model.g.dart:
```diff
@@ -0,0 +1,116 @@
+// GENERATED CODE - DO NOT MODIFY BY HAND
```
auto generated code ftw

efabens commented on lib/plugin_tozny.dart:
```diff
@@ -11,4 +12,46 @@ class PluginTozny {
     return version;
   }
+
+  static Future<Record> readRecord(String recordID, ClientCredentials creds) async {
```
Flutter by design is object-oriented. I really don't like that all of these methods are static. I realize this is just trying to get code out the door but there should really be a or tozny class that is instantiated and does the rest of these methods not needing to pass in a client credential.

willmichael replied:
Agreed and will refactor here, I let usability/design slip in the "auto port all the methods" mode

efabens reviewed (commented):
I don't think that the static methods are the correct approach for this plugin. There should be a tozny object that when instantiated does things. If that refactor is going to cause a deadline to be missed then you have my reluctant approval.

galxy25 commented:
> I don't think that the static methods are the correct approach for this plugin. There should be a tozny object that when instantiated does things. If that refactor is going to cause a deadline to be missed then you have my reluctant approval.

@willmichael talk to @efabens , if you two agree the best long term approach is the OO way, the client deliverable can wait until the necessary refactoring occurs. For our current customer relationship and need the bar is low for the scope of functionality we need to deliver, but the quality of that functionality needs to still meet our internal standards

willmichael committed:
Misc cleanup

Rename package identifier
Bump java-sdk versioning
Update dev doc and changelog
Move plugin structure to OO

willmichael committed:
remove local settings files
